compiler/rustc_data_structures/src/thin_vec.rs

```diff
@@ -121,6 +123,20 @@ impl<T: HashStable<CTX>, CTX> HashStable<CTX> for ThinVec<T> {
     }
 }
 
+impl<T: Hash> Hash for ThinVec<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        (**self).hash(state)
+    }
+}
+
+impl<T: PartialEq> PartialEq for ThinVec<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.deref().eq(other.deref())
+    }
+}
+
+impl<T: PartialEq> Eq for ThinVec<T> {}
```

Comment:
Why not just `derive(Hash, PartialEq, Eq)`?

Reply:
It didn't compile with Derive, but I suspect it was a fluke, I'll try again.